examples/chatgpt/rag-quickstart/azure/Azure_AI_Search_with_Azure_Functions_and_GPT_Actions_in_ChatGPT.ipynb

```diff
@@ -0,0 +1,1330 @@
+                "    \n",
+                "    category = categorize_text(' '.join(content_text), categories)\n",
+                "    print(f\"Categorized {file_name} as {category}\")\n",
+                "    \n",
+                "    # Prepare the data to be appended\n",
+                "    data = []\n",
+                "    for i, content_vector in enumerate(content_vectors):\n",
+                "        data.append({\n",
+                "            \"id\": f\"{idx}_{i}\",\n",
+                "            \"vector_id\": f\"{idx}_{i}\",\n",
+                "            \"title\": title_text[0],\n",
+                "            \"text\": content_text[i],\n",
+                "            \"title_vector\": json.dumps(title_vectors[0]),  # Assuming title is short and has only one chunk\n",
+                "            \"content_vector\": json.dumps(content_vector),\n",
+                "            \"category\": category\n",
+                "        })\n",
+                "        print(f\"Appended data for chunk {i + 1}/{len(content_vectors)} of {file_name}\")\n",
+                "    \n",
+                "    return data\n",
+                "\n",
+                "def process_files(folder_name):\n",
+                "    # List all .txt and .pdf files in the specified folder\n",
+                "    files = [os.path.join(folder_name, f) for f in os.listdir(folder_name) if f.endswith('.txt') or f.endswith('.pdf')]\n",
+                "    data = []\n",
+                "\n",
+                "    # Process each file concurrently\n",
+                "    with concurrent.futures.ThreadPoolExecutor() as executor:\n",
+                "        futures = {executor.submit(process_file, file_path, idx, categories, embeddings_model): idx for idx, file_path in enumerate(files)}\n",
+                "        for future in concurrent.futures.as_completed(futures):\n",
+                "            try:\n",
+                "                result = future.result()\n",
+                "                data.extend(result)\n",
+                "            except Exception as e:\n",
+                "                print(f\"Error processing file: {str(e)}\")\n",
+                "\n",
+                "    # Write the data to a CSV file\n",
+                "    csv_file = os.path.join(\"..\", \"embedded_data.csv\")\n",
+                "    with open(csv_file, 'w', newline='', encoding='utf-8') as csvfile:\n",
+                "        fieldnames = [\"id\", \"vector_id\", \"title\", \"text\", \"title_vector\", \"content_vector\",\"category\"]\n",
+                "        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)\n",
+                "        writer.writeheader()\n",
+                "        for row in data:\n",
+                "            writer.writerow(row)\n",
+                "            print(f\"Wrote row with id {row['id']} to CSV\")\n",
+                "\n"
```

Comment:
Same as previous comment; let's consolidate the comments to help with readability.

Do we also need all of these print statements?